pkg/pcr/get_measurements.go

```diff
@@ -53,3 +52,133 @@ func GetMeasurements(
 	}
 	return
 }
+
+func getMeasurements(
+	pcrID ID,
+	firmware Firmware,
+	config MeasurementConfig,
+) (
+	Measurements,
+	Flow,
+	map[string]interface{},
+	error,
+) {
+	resultConfig := config
+	debugInfo := map[string]interface{}{
+		"config_orig":   config,
+		"config_result": &resultConfig,
+	}
+
+	// Detect attestation flow
+	detectedFlow, detectFlowErr := DetectAttestationFlow(firmware, config.Registers, config.TPMDevice)
+	if detectedFlow != FlowAuto {
+		debugInfo["detectedAttestationFlow"] = detectedFlow.String()
+	}
+
+	if resultConfig.Flow == FlowAuto {
+		if detectedFlow == FlowAuto {
+			return nil, FlowAuto, nil,
+				fmt.Errorf("unable to detect attestation flow (try option manually selecting flow using --flow option): %w", detectFlowErr)
+		}
+		resultConfig.Flow = detectedFlow
+	}
+
+	// Collect measurements
+	measurements, warnings, err := newMeasurementsCollector(firmware).CollectMeasurements(pcrID, resultConfig)
+	if err != nil {
+		err = fmt.Errorf("unable to collect measurements: %w", err)
+	}
+	debugInfo["warnings"] = (&errors.MultiError{}).Add(warnings, detectFlowErr).ReturnValue()
+
+	return measurements, resultConfig.Flow, debugInfo, err
+}
+
+type measurementsCollector struct {
+	firmware         Firmware
+	fitEntriesResult *[]fit.Entry
+	pcdDataResult    *pcd.ParsedFirmware
+	amdFirmware      *amd.AMDFirmware
+	errors           errors.MultiError
+	warnings         errors.MultiError
+}
+
+func newMeasurementsCollector(firmware Firmware) *measurementsCollector {
+	return &measurementsCollector{
+		firmware: firmware,
+	}
+}
+
+func (c *measurementsCollector) Firmware() Firmware {
+	return c.firmware
+}
+
+func (c *measurementsCollector) FITEntries() []fit.Entry {
+	if c.fitEntriesResult != nil {
+		return *c.fitEntriesResult
+	}
+
+	fitEntries, err := c.firmware.GetFIT()
+	if err != nil {
+		_ = c.errors.Add(ErrGetFIT{Err: err})
```

Comment:
nit

make it as a method of measurementsCollector:

func (c *measurementsCollector) addError(err error)

Reply:
Yeah. Thanks for the advice. I'll continue changing this code in another PR, hoping won't forget to fix this as well :)